### PR TITLE
Scene Manager Prep

### PR DESCRIPTION
**Pull request by michaelosbornegit**
1. Plane extends ship (but have not experimented with taking plane control away from user yet)

2. Refactored Jim's showMessage to allow for dynamic message displays, this will be used for scene manager messages like 'wave complete' 'get ready' 'wave 3'

3. prototype of scene manager complete, needs much more functionality and testing but for right now it's ready to go live and will be updated later today

Sorry I touched so much in this PR, let me know if you have a hard time resolving conflicts. I realize the mistake and will separate tasks in the future.

### CODE REVIEW

File: gameengine.js [hunk 1]
```diff
@@ -154,22 +149,22 @@ class GameEngine {
...
-    
+
       this.stats.update(); // remove for production
     }
   }
 
   pause() {
     if (this.isPaused) {
-      hideMessage('pause-message');
+      hideMessage();
```
Comment (jim-rosales) [inline gameengine.js]: Just pulled and the game doesn't resume after pressing p again, does it on your branch?

Comment (michaelosbornegit) [reply gameengine.js]: pull again, I snuck in a fix

**Comment by jim-rosales** [comment]
Looks good to me! Great job!

**Comment by jim-rosales** [comment]
Lol accidentally closed the PR

**Comment by codemalone** [comment]
this looks good to me. I will begin from this commit to add multi-ring support to the Ship.

**Comment by codemalone** [comment]
On second look, I did make a small change to how Plane extends Ship. I put the custom plane update behavior into updateHelm(). This way it overrides the Ship methods for 
  updateHelm()
  updateCollisionDetection()
but not the Ship.update() method. This will support helm override and identical weapon behavior at the Ship level. No other changes here.

**Pull request merged**